web/src/custom-typings/window/index.d.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare global {
+  interface Window {
+    microapps: any;
+  }
+}
+
+// Export nothing.
+// Added because we can only `declare global` in files that has import or export.
```

Comment:
Why do we want to declare global when we are not exporting it?

Reply:
@xlai20 We don't have to export global because TS already have its own globals, so by declaring it like this, TS will merge it with the existing globals.
`Window` is an existing global type and what we are doing here is just extending it.
Can also refer to: https://www.typescriptlang.org/docs/handbook/declaration-files/templates/global-modifying-module-d-ts.html